File: template/multi-platform-js/Classes/RootWindow.cpp


#include "RootWindow.h"
#include "FirstViewController.h"

RootWindow* RootWindow::create()
{
    RootWindow* _window = new RootWindow();
    _window->init();
    _window->autorelease();
    return _window;
}

RootWindow::RootWindow()
{

}

RootWindow::~RootWindow()
{

}

bool RootWindow::init()
{
    if (!CAWindow::init())
    {
        return false;
    }
    
    FirstViewController* _viewController = new FirstViewController();
    _viewController->init();
    this->setRootViewController(_viewController);
    _viewController->release();
    
    return true;
}

File: template/multi-platform-js/Resources/js/root_window.js
require('js/first_view_controller.js');

var RootWindow = ca.CAWindow.extend(
    {
        ctor: function () {
            this._super();

            var firstViewController = new FirstViewController();
            firstViewController.init();
            this.setRootViewController(firstViewController);
            firstViewController.release();
        }
    }
);
